File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        int len = strs.size();

        unordered_map<string, vector<string>> groups;
        for (int i = 0; i < len; i++) {
            string temp = strs[i];
            sort(temp.begin(), temp.end());

            groups[temp].push_back(strs[i]);
        }

        vector<vector<string>> res;
        for (auto el : groups) {
            res.push_back(el.second);
        }

        return res;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        int n = strs.length;

        Map<String, List<String>> mpp = new HashMap<>();
        for (String str : strs) {
            char[] strChars = str.toCharArray();
            Arrays.sort(strChars);
            String sortedStr = new String(strChars);
            List<String> anagrams = mpp.getOrDefault(sortedStr, new ArrayList<String>());
            anagrams.add(str);
            mpp.put(sortedStr, anagrams);
        }

        List<List<String>> res = new ArrayList<>();
        mpp.forEach((k, v) -> {
            res.add(v);
        });

        return res;
    }
}